asyncpg/connect_utils.py

```diff
@@ -39,46 +62,44 @@ class SSLMode(enum.IntEnum):
     verify_full = 5
 
     @classmethod
-    def parse(cls, sslmode):
+    def parse(
+        cls: typing.Type[_SSLMode],
+        sslmode: typing.Union[str, _SSLMode]
+    ) -> _SSLMode:
         if isinstance(sslmode, cls):
             return sslmode
-        return getattr(cls, sslmode.replace('-', '_'))
-
-
-_ConnectionParameters = collections.namedtuple(
-    'ConnectionParameters',
-    [
-        'user',
-        'password',
-        'database',
-        'ssl',
-        'sslmode',
-        'connect_timeout',
-        'server_settings',
-    ])
+        return typing.cast(
+            _SSLMode,
+            getattr(cls, typing.cast(str, sslmode).replace('-', '_'))
+        )
 
 
-_ClientConfiguration = collections.namedtuple(
-    'ConnectionConfiguration',
-    [
-        'command_timeout',
-        'statement_cache_size',
-        'max_cached_statement_lifetime',
-        'max_cacheable_statement_size',
-    ])
+class _ConnectionParameters(typing.NamedTuple):
+    user: str
+    password: typing.Optional[str]
+    database: str
+    ssl: typing.Optional[_ParsedSSLType]
+    sslmode: typing.Optional[SSLMode]
+    connect_timeout: float
+    server_settings: typing.Optional[typing.Dict[str, str]]
 
 
-_system = platform.uname().system
+class _ClientConfiguration(typing.NamedTuple):
+    command_timeout: typing.Optional[float]
+    statement_cache_size: int
+    max_cached_statement_lifetime: int
+    max_cacheable_statement_size: int
 
 
-if _system == 'Windows':
-    PGPASSFILE = 'pgpass.conf'
-else:
-    PGPASSFILE = '.pgpass'
+_system: typing_extensions.Final = platform.uname().system
+PGPASSFILE: typing_extensions.Final = (
+    'pgpass.conf' if _system == 'Windows' else '.pgpass'
+)
 
 
-def _read_password_file(passfile: pathlib.Path) \
-        -> typing.List[typing.Tuple[str, ...]]:
+def _read_password_file(
+    passfile: pathlib.Path
+) -> typing.List[typing.Tuple[str, ...]]:
```

Comment:
Why is this style not used everywhere? Is line breaks with `\` better?

Reply:
There seems to be a mix of styles in the codebase. I use the black style (seen here) in my own projects, and I saw it in some of the newer code, so I went with it. If that needs to be changed, I can format the code a different way.

Reply:
I prefer Black style, but it is up to asyncpg maintainers